pilot/pkg/autoregistration/controller.go

```diff
@@ -651,6 +651,10 @@ func workloadEntryFromGroup(name string, proxy *model.Proxy, groupCfg *config.Co
 	// state is not fully initialized. Therefore, we check the bootstrap node.
 	if proxy.XdsNode.Locality != nil {
 		entry.Locality = util.LocalityToString(proxy.XdsNode.Locality)
+		// the label has been converted to "istio-locality: region/zone/subzone"
+		// in pilot/pkg/xds/ads.go, and `/` is not allowed in k8s label value.
+		// Instead of converting again, we delete it since has set WorkloadEntry.Locality
+		delete(entry.Labels, model.LocalityLabel)
```

Comment:
Why do we have to do it here? Why not do at after L639. We do not use that label any more?

Also I think proxy.XdsNode.Locality would never be nil - right?

Reply:
As it is duplicated with WLE.Locality field, so i decided to delete it instead of converting again, whatever the label on wle will not be used.

Reply:
>Why do we have to do it here? Why not do at after L639. We do not use that label any more?

Not a must


> Also I think proxy.XdsNode.Locality would never be nil - right?

I think it can be nil, if proxy bootstrap does not set it.  Thanks for this question, I think we should not proxy.Locality instead here, because it may be inferred from meta labels